pkgs/development/libraries/indicator-application/gtk3.nix

```diff
@@ -1,39 +1,36 @@
-{ stdenv, fetchurl, lib, file
-, pkgconfig, autoconf
+{ stdenv, fetchbzr
+, pkgconfig, autoreconfHook, systemd
 , glib, dbus-glib, json-glib
 , gtk3, libindicator-gtk3, libdbusmenu-gtk3, libappindicator-gtk3 }:
 
-with lib;
-
 stdenv.mkDerivation rec {
-  name = "indicator-application-gtk3-${version}";
-  version = "${versionMajor}.${versionMinor}";
-  versionMajor = "12.10";
-  versionMinor = "0";
-
-  src = fetchurl {
-    url = "${meta.homepage}/${versionMajor}/${version}/+download/indicator-application-${version}.tar.gz";
-    sha256 = "1z8ar0k47l4his7zvffbc2kn658nid51svqnfv0dms601w53gbpr";
+  pname = "indicator-application";
+  version = "12.10.1";
+
+  name = "${pname}-gtk3-${version}";
+
+  src = fetchbzr {
+    url = "https://code.launchpad.net/~indicator-applet-developers/${pname}/trunk.17.04";
+    rev = "260";
+    sha256 = "1f0jdyqqb5g86zdpbcyn16x94yjigsfiv2kf73dvni5rp1vafbq1";
   };
 
-  nativeBuildInputs = [ pkgconfig autoconf ];
+  nativeBuildInputs = [ pkgconfig autoreconfHook systemd ];
 
   buildInputs = [
     glib dbus-glib json-glib
     gtk3 libindicator-gtk3 libdbusmenu-gtk3 libappindicator-gtk3
   ];
 
   postPatch = ''
-    substituteInPlace configure.ac \
-      --replace 'DBUSSERVICEDIR=`$PKG_CONFIG --variable=session_bus_services_dir dbus-1`' \
-                "DBUSSERVICEDIR=$out/share/dbus-1/services"
-    autoconf
-    for f in {configure,ltmain.sh,m4/libtool.m4}; do
-      substituteInPlace $f \
-        --replace /usr/bin/file ${file}/bin/file
-    done
-    substituteInPlace src/Makefile.in \
+    substituteInPlace data/Makefile.am \
+      --replace "/etc/xdg/autostart" "$out/etc/xdg/autostart"
+
+    substituteInPlace src/Makefile.am \
       --replace 'applicationlibdir = $(INDICATORDIR)' "applicationlibdir = $out/lib"
+
+    substituteInPlace configure.ac \
+      --replace 'SYSTEMD_USERDIR=`$PKG_CONFIG --variable=systemduserunitdir systemd`' 'SYSTEMD_USERDIR="$out/lib/systemd/user"'
```

Comment:
Can you try setting the `PKG_CONFIG_SYSTEMD_SYSTEMDUSERUNITDIR = "$(out)/lib/systemd/user";` attribute in the expression?